### PR TITLE
[Imaging Uploader / Server Process Manager] Fix invalid upload state (RM #14927)

### PR DESCRIPTION
**Pull request by davidblader**
https://redmine.cbrain.mcgill.ca/issues/14927

@nicolasbrossard 
I still was not able to replicate the error exactly, but i added the check to make sure `imaging_upload_file.pl` was on the system before proceeding with the upload / insertion as we discussed
let me know if this addresses the issue you encountered

### CODE REVIEW

**Comment by ridz1208** [comment]
@cmadjar @MounaSafiHarab would you be able to test this ?

**Comment by ridz1208** [comment]
@davidblader failing travis tests, issue in server process manager

File: modules/imaging_uploader/php/imaging_uploader.class.inc [hunk 1]
```diff
@@ -155,15 +155,24 @@ class Imaging_Uploader extends \NDB_Menu_Filter_Form
...
-                // its namespace classes
-                $spm_module = \Module::factory('server_processes_manager');
-                // Perform the real upload on the server
-                $serverProcessLauncher = new SP\ServerProcessLauncher();
-                $serverProcessLauncher->mriUpload(
-                    $this->mri_upload_id,
-                    $this->uploaded_file_path
-                );
+                try {
+                    // Instanciate the server process module to autoload
```
Comment (HenriRabalais) [inline modules/imaging_uploader/php/imaging_uploader.class.inc]: small typo: `intanciate` -> `instantiate`

**Comment by davidblader** [comment]
assuming this is failing travis because of the exception i've added gets thrown via
`ServerProcessesManager -> ServerProcessesMonitor -> ServerProcessFactory -> MriUploadServerProcess`
@driusan @kongtiaowang any ideas for how to circumvent travis failing here? could we put the LORIS-MRI code on the docker container / whatever environment travis is running on?

**Comment by ridz1208** [comment]
@cmadjar can this be closed ?

**Comment by davidblader** [comment]
> assuming this is failing travis because of the exception i've added gets thrown via
> `ServerProcessesManager -> ServerProcessesMonitor -> ServerProcessFactory -> MriUploadServerProcess`
> @driusan @kongtiaowang any ideas for how to circumvent travis failing here? could we put the LORIS-MRI code on the docker container / whatever environment travis is running on?

reminder that this will probably fail travis eternally

File: modules/server_processes_manager/php/mriuploadserverprocess.class.inc [hunk 2]
```diff
@@ -112,6 +112,13 @@ class MriUploadServerProcess extends AbstractServerProcess
             );
         }
 
+        // throw error if unable to locate insertion script
+        if (!file_exists($this->_mriCodePath . self::IMAGING_UPLOAD_FILE_PATH)) {
```
Comment (nicolasbrossard) [inline modules/server_processes_manager/php/mriuploadserverprocess.class.inc]: I believe the check for the presence of the imaging upload script is not necessary. An exception will be thrown anyway if an attempt to start this process is made and the script does not exist.

**Comment by cmadjar** [comment]
@nicolasbrossard @davidblader I tested the PR again using a false MRICode path and it fails silently. I can upload the scan successfully however, there is no error message anywhere telling me that the MRI scripts can not be found. In the log viewer, it only says "MRI pipeline not yet executed for this upload" while the autolaunch was set to "Yes" and I was expecting an error message saying the scripts cannot be found at least in the log window. 

Where is the exception thrown? How can it be displayed in the Log Viewer?

**Comment by nicolasbrossard** [comment]
[https://github.com/aces/Loris/pull/4495](https://github.com/aces/Loris/pull/4495) does what this PR is supposed to do: closing.

**Pull request closed without merging**